operator/controllers/clusterconfig_controller.go

```diff
@@ -47,15 +58,72 @@ type ClusterConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *ClusterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	reqLogger := log.WithValues("Request.Name", req.Name)
+	reqLogger.Info("reconciling ClusterConfig: START")
+	defer reqLogger.Info("reconciling ClusterConfig: STOP")
+
+	instance := &metalk8sscalitycomv1alpha1.ClusterConfig{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if req.Name == instanceName {
+				panic(fmt.Errorf(
+					"%s ClusterConfig object should not be deleted", req.Name,
+				))
```

Comment:
Worth adding a comment about the operator behavior here (either directly in the message, or at least in the code): it's not obvious that the panic will lead to a restart (because, K8s) followed by a re-creation of the main ClusterConfig on startup.

Reply:
Ok I will add a comment

Reply:
Done